### DISCUSSION TITLE
Improving agarose gels to increase readability

### DISCUSSION THREAD

**Discussion by kcorreia** [pydna-group/pydna#98] [Ideas]
- allowing sample ids to be shown on gel or longer names (maybe rotated for longer names)
- invert colours (I see that there is code commented out right now) which is better for printing
- having mass/weights applied to fragments that make some fragments bolder than others. For example, NEB ladder has most fragments with 30-60 ng, except 500bp, 1000bp, 3000bp with 120ng.

![image](https://user-images.githubusercontent.com/11604330/218765534-75d52550-78b6-44d5-a0d8-04d1c505b6f7.png)

**Comment by BjornFJohansson** [comment]
The first point is a great suggestion. The code would need to make sensible choices as to the placement and text size. Any ideas about that? Perhaps some code I could reuse. 

The second is easy:
```
im = gel([ GeneRuler_1kb_plus, [band, ]])
from PIL import ImageOps
ImageOps.invert(im)
```
This could be made into a method, but I thought it might be simpler as is.

The third should be there already. Look at the ladders.py module.

**Comment by kcorreia** [comment]
I'm not sure on how to implement text with PIL but soon enough I'll get frustrated with manually annotating printouts and learn more about PIL.

**Comment by BjornFJohansson** [comment]
Simply adding the size below the band is easy. The problem is what to do when the bands are in close proximity.
Have a look at BandWitch from Edinburgh https://github.com/Edinburgh-Genome-Foundry/BandWitch
It has a sort of gel simulator for schematic gels.

https://www.blog.pythonlibrary.org/2021/02/02/drawing-text-on-images-with-pillow-and-python
